Jenkinsfile

```diff
@@ -11,39 +13,47 @@ pipeline {
     VIEW = getViewName(REPO)
   }
   options {
+    buildDiscarder(xmosDiscardBuildSettings())
     skipDefaultCheckout()
     timestamps()
-    // on develop discard builds after a certain number else keep forever
-    buildDiscarder(logRotator(
-        numToKeepStr:         env.BRANCH_NAME ==~ /develop/ ? '100' : '',
-        artifactNumToKeepStr: env.BRANCH_NAME ==~ /develop/ ? '100' : ''
-    ))
   }
+  parameters {
+    string(
+      name: 'TOOLS_VERSION',
+      defaultValue: '15.3.0',
+      description: 'The XTC tools version'
+    )
+  }
+
   stages {
     stage('Get view') {
       steps {
         xcorePrepareSandbox("${VIEW}", "${REPO}")
       }
     }
-    stage('Library checks') {
-      steps {
-        xcoreLibraryChecks("${REPO}")
-      }
-    }
-    stage('xCORE builds') {
+    stage('Build examples') {
       steps {
+        println "Stage running on ${env.NODE_NAME}"
         dir("${REPO}") {
-          // xcoreAllAppsBuild('examples')
-          xcoreAllAppNotesBuild('examples')
-          dir("${REPO}") {
-            runXdoc('doc')
+          checkout scm
+
+          dir("examples") {
+            withTools(params.TOOLS_VERSION) {
+              sh 'cmake -G "Unix Makefiles" -B build'
+              sh 'xmake -C build -j 8'
```

Comment:
Small thing but seems most of the linux agents have 16 processors / 32 threads.

Reply:
This was running on macOS (not sure why). I'll try changing to Linux agent and remove [jobs] so it can use what it likes.

Reply:
Moving to linux doesn't do much for the build step (it's only ~30 seconds) , but I can push the test step (--numprocesses) to reduce from 30 to 10 mins.